### PR TITLE
Adds e2e test for lab.prompt and use React to render the prompt

### PR DESCRIPTION
**Pull request by kamilkisiela**
I changed the `window.prompt` to custom React component to avoid passing `allow-modal` to iframe's sandbox.
Screenshot 2024-12-31 at 15 01 06
Screenshot 2024-12-31 at 15 01 20
Screenshot 2024-12-31 at 14 56 27

### CODE REVIEW

File: cypress.config.ts [hunk 1]
```diff
@@ -16,6 +16,10 @@ export default defineConfig({
   video: isCI,
   screenshotOnRunFailure: isCI,
   defaultCommandTimeout: 15_000, // sometimes the app takes longer to load, especially in the CI
+  // We need to disable this to stub window.prompt - cross-origin iframes
+  // The prompt is accessible and displayed to the user, but the test runner can't interact with it,
+  // meaning we can't automate the input of the prompt.
+  chromeWebSecurity: false,
```
Comment (kamilkisiela) [inline cypress.config.ts:22]: @n1ru4l @dotansimha

File: packages/web/app/src/lib/preflight-sandbox/graphiql-plugin.tsx [hunk 2]
```diff
@@ -316,11 +316,12 @@ export function usePreflightScript(args: {
         src="/__preflight-embed"
         title="preflight-worker"
         className="hidden"
+        data-cy="preflight-embed-iframe"
         /**
          * In DEV we need to use "allow-same-origin", as otherwise the embed can not instantiate the webworker (which is loaded from an URL).
          * In PROD the webworker is not
          */
-        sandbox={import.meta.env.DEV ? 'allow-scripts allow-same-origin' : 'allow-scripts'}
+        sandbox={'allow-scripts allow-modals' + (import.meta.env.DEV ? ' allow-same-origin' : '')}
```
Comment (kamilkisiela) [inline packages/web/app/src/lib/preflight-sandbox/graphiql-plugin.tsx:324]: @n1ru4l @dotansimha

File: cypress/e2e/preflight-script.cy.ts [hunk 3]
```diff
@@ -98,6 +100,63 @@ throw new TypeError('Test')`,
     );
   });
 
+  it('prompt and pass the awaited response', () => {
+    cy.viewport(1600, 800);
```
Comment (kamilkisiela) [inline cypress/e2e/preflight-script.cy.ts:104]:
```suggestion
```

File: cypress/e2e/app.cy.ts [hunk 4]
```diff
@@ -1,9 +1,5 @@
 import { generateRandomSlug, getUserData } from '../support/testkit';
 
-Cypress.on('uncaught:exception', (_err, _runnable) => {
-  return false;
-});
-
```
Comment (saihaj) [inline cypress/e2e/app.cy.ts:6]: why remove this?

Comment (kamilkisiela) [reply cypress/e2e/app.cy.ts:6]: it's moved to `support/e2e`

**Review by saihaj — APPROVED** [review]

**Pull request merged**